core/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramParser.java

```diff
@@ -52,6 +51,8 @@ protected DateHistogramAggregationBuilder createFactory(String aggregationName,
             factory.interval((Long) interval);
         } else if (interval instanceof DateHistogramInterval) {
             factory.dateHistogramInterval((DateHistogramInterval) interval);
+        } else {
+            throw new AssertionError();
```

Comment:
I think it is a bit strange to throw an AssertionError when assertions might not be enabled? Maybe just throw and IllegalArgumentException or a RuntimeException or ElasticsearchException or something?